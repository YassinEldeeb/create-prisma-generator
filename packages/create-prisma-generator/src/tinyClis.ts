// No need to attach @latest at the end of the package name cause
// npx will always execute the latest available version in the NPM registry
export const CLIs = {
  typescriptTemplate(path: string) {
    return `npx @cpg-cli/template-typescript ${path}`
  },
  rootConfigs(path: string) {
    return `npx @cpg-cli/root-configs ${path}`
  },
  usageTemplate(path: string) {
    return `npx @cpg-cli/template-gen-usage ${path}`
  },
  javascriptTemplate(path: string) {
    return `npx @cpg-cli/template ${path}`
  },
  githubActionsTemplate(path: string) {
    return `npx @cpg-cli/github-actions ${path}`
  },
  setupSemanticRelease(path: string, workspaceFlag: string) {
    return `npx @cpg-cli/semantic-releases ${path} ${workspaceFlag}`
  },
}
