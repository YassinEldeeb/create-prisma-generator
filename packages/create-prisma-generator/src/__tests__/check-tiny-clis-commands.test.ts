import { CLIs } from '../tinyClis'
import { serializeCLI } from './__helpers__/snapshotSerializers'

Object.keys(CLIs).forEach((cli) => {
  test(`${cli} command is configured properly`, () => {
    expect(serializeCLI(cli as keyof typeof CLIs)).toMatchSnapshot()
  })
})
