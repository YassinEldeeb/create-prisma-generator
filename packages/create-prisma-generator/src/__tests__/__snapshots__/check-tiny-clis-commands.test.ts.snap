// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`githubActionsTemplate command is configured properly 1`] = `"npx @cpg-cli/github-actions path"`;

exports[`javascriptTemplate command is configured properly 1`] = `"npx @cpg-cli/template path"`;

exports[`rootConfigs command is configured properly 1`] = `"npx @cpg-cli/root-configs path"`;

exports[`setupSemanticRelease command is configured properly 1`] = `"npx @cpg-cli/semantic-releases path workspace"`;

exports[`typescriptTemplate command is configured properly 1`] = `"npx @cpg-cli/template-typescript path"`;

exports[`usageTemplate command is configured properly 1`] = `"npx @cpg-cli/template-gen-usage path"`;
